cpp/include/cudf/strings/string_view.inl

```diff
@@ -84,6 +88,16 @@ __device__ inline size_type string_view::length() const
 {
     if( _length <= UNKNOWN_STRING_LENGTH )
         _length = strings::detail::characters_in_string(_data,_bytes);
+    if( _length && (_char_width <= UNKNOWN_CHAR_WIDTH) )
+    {
+        const BYTE* bytes = reinterpret_cast<const BYTE*>(data());
+        auto chwidth = bytes_in_utf8_byte(*bytes); // see if they are all the same width
+        _char_width = (thrust::all_of( thrust::seq, bytes, bytes + size_bytes(), 
```

Comment:
(1.) AFAIK, you don't need `__device__` qualifier for the nested lambda called from already a `__device__` function via `thrust::seq`.
(2.) You might consider replacing `all_of(...,predicate)` with `find_if(..., !predicate)` (negated). It would be faster, most of the time (regardless of some "rumors" I've heard that `all_of()` might be a bit slow, on occasion).